Parsedown.php

```diff
@@ -1485,10 +1503,80 @@ function parse($text)
         return $markup;
     }
 
+    protected function sanitiseElement(array $Element)
+    {
+        static $badAttributeChars = "\"'= \t\n\r\0\x0B";
+        static $safeUrlNameToAtt  = array(
+            'a'   => 'href',
+            'img' => 'src',
+        );
+
+        if (isset($safeUrlNameToAtt[$Element['name']]))
+        {
+            $Element = $this->filterUnsafeUrlInAttribute($Element, $safeUrlNameToAtt[$Element['name']]);
+        }
+
+        if ( ! empty($Element['attributes']))
+        {
+            foreach ($Element['attributes'] as $att => $val)
+            {
+                # clear out nulls
+                if ($val === null)
+                {
+                    unset($Element['attributes'][$att]);
+                }
+                # filter out badly parsed attribute
+                elseif (strpbrk($att, $badAttributeChars) !== false)
+                {
+                    unset($Element['attributes'][$att]);
+                }
+            }
+
+            $onEventAttributeKeys = preg_grep('/^on/i', array_keys($Element['attributes']));
+
+            foreach ($onEventAttributeKeys as $att)
+            {
+                unset($Element['attributes'][$att]);
+            }
+        }
+
+        return $Element;
+    }
+
+    protected function filterUnsafeUrlInAttribute(array $Element, $attribute)
+    {
+        if ($this->safeLinksEnabled)
+        {
+            $safe = false;
+
+            foreach ($this->safeLinksWhitelist as $scheme)
+            {
+                if (stripos($Element['attributes'][$attribute], $scheme) === 0)
```

Comment:
IMO we should make an effort to expand the list of allowed protocols as much as possible and ones that we don't know about can be added as bugfixes as we go.

And forgetting to support something like `steam://` IMO is more favourable to accidently letting someone's website become vulnerable to XSS because we didn't get the filter quite right, especially when a user could easily just leave the URL as plain text as a work around.